pinecone/management_client.go

```diff
@@ -0,0 +1,488 @@
+package pinecone
+
+import (
+	"context"
+	"fmt"
+	"github.com/deepmap/oapi-codegen/v2/pkg/securityprovider"
+	"github.com/google/uuid"
+	"github.com/pinecone-io/go-pinecone/internal/gen/management"
+	"net/http"
+)
+
+// ManagementClient provides a high-level interface for interacting with the
+// Pinecone management plane API. It encapsulates the necessary authentication,
+// request creation, and response handling for the API's operations.
+//
+// The ManagementClient is designed to simplify the management of projects and
+// their API keys by abstracting away the direct handling of HTTP requests and
+// responses. It leverages a generated low-level client (`restClient`) for
+// communication with the API, ensuring that requests are properly authenticated
+// and formatted according to the API's specification.
+//
+// Fields:
+//   - apiKey: The API key used for authenticating requests to the management API.
+//     This key should have the necessary permissions for the operations you intend to perform.
```

Comment:
it's tough because org keys aren't generally available yet... so there's nothing to point to publicly. I'm open to suggestions. I think this general confusion exists when discussing the Management API in general.